octoprint_octorelay/__init__.py

```diff
@@ -209,11 +210,20 @@ def handle_plugin_event(self, event, scope = None):
         if needs_ui_update:
             self.update_ui() # issue 190
 
+    def is_printer_relay(self, index) -> bool:
+        printer_relay = self._settings.get(["common", "printer"], merged=True) # expensive
+        return printer_relay is not None and printer_relay == index
+
     def toggle_relay(self, index, target: Optional[bool] = None):
         settings = self._settings.get([index], merged=True) # expensive
         if not bool(settings["active"]):
             self._logger.debug(f"Refusing to switch the relay {index} since it's disabled")
             return
+        if self.is_printer_relay(index):
+            self._logger.debug(f"{index} is the printer relay")
+            if self._printer.is_operational():
+                self._logger.debug(f"Disconnecting from the printer before turning {index} OFF")
```

Comment:
the `target` state can actually be `None` meaning the toggling.
the relay's current state is unknown at this point as well.
therefore I'm intentionally referring to the printer's state, which makes more sense for the action of disconnecting, so the target relay state is assumed from the printer's state here.

Reply:
However, I should exclude the case when the `target` is `True`.
Can be some postponed task which leads to turning the relay even when it's already on. ⚠️

Reply:
done in 46e75f0